metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -260,15 +261,39 @@ class Compilers(
           AdjustedLspData.default,
         )
 
+        val (prependedLinesSize, modifiedText) = Option
+          .when(path.isSbt)(
+            buildTargets
+              .sbtAutoImports(path)
+          )
+          .flatten
+          .fold((0, input.value))(imports =>
+            (
+              imports.size,
+              SbtBuildTool.prependAutoImports(input.value, imports),
+            )
+          )
+
         outlineFilesProvider.didChange(pc.buildTargetId(), path)
 
         for {
           ds <-
             pc
               .didChange(
-                CompilerVirtualFileParams(path.toNIO.toUri(), input.value)
+                CompilerVirtualFileParams(path.toNIO.toUri, modifiedText)
               )
               .asScala
+              .map(diagnosticsList => {
+                diagnosticsList.forEach(diagnostic => {
```

Comment:
We should be able to use `adjust` for this I think, which already translates positions for sbt files